src/GitVersionExe/HelpWriter.cs

```diff
@@ -53,8 +53,13 @@ public void WriteTo(Action<string> writeAction)
  # AssemblyInfo updating
     /updateassemblyinfo
                     Will recursively search for all 'AssemblyInfo.cs' files in the git repo and update them
-    /updateassemblyinfofilename
```

Comment:
Hm, with `updateassemblyinfofilename` changed to `updatefilenames`, this is a breaking change and will have to wait for v6 anyway. Perhaps leave `updateassemblyinfofilename` as is and introduce another parameter that serves the same function, but for the `updateprojectfiles` parameter? I know it's messy, but this will all be cleaned up in v6.

Reply:
How many a developer has been fooled by the phrase _it's messy but we'll get it all cleaned up next version.._ 😄 

Will do.

Reply:
@asbjornu So uhh..I'm reverting some of the _naming_ I did, but it appears that that parameter is not implemented, and the only way to pass file names to UpdateAssemblyInfo (and now UpdateProjectInfo) is to specify filenames _after_ the update switch. Realistically that parameter can be removed without breaking anything since _including_ it seems to break things in and of itself.

In any case give it another look and see if I unbroke'd it.

Reply:
Example of a test that fails -
```cs
[Test]
public void UpdateAssemblyInfoWithFilenameFromUpdateAssemblyInfoFilenameSwitch()
{
    using var repo = new EmptyRepositoryFixture();

    var assemblyFile = Path.Combine(repo.RepositoryPath, "CommonAssemblyInfo.cs");
    using var file = File.Create(assemblyFile);

    var arguments = argumentParser.ParseArguments($"-targetpath {repo.RepositoryPath} -updateAssemblyInfo -updateassemblyinfofilename CommonAssemblyInfo.cs");
    arguments.UpdateAssemblyInfo.ShouldBe(true);
    arguments.UpdateAssemblyInfoFileName.Count.ShouldBe(1);
    arguments.UpdateAssemblyInfoFileName.ShouldContain(x => Path.GetFileName(x).Equals("CommonAssemblyInfo.cs"));
}       
```

Reply:
Provided everything I said is true - would you be comfortable mimicking `/updateAssemblyInfo` and relegating passing Filenames to just that switch, rather than implementing something for project files that isn't in assembly files?

Reply:
Good find, @svengeance. Indeed, it looks like you're right and that `updateassemblyinfofilename` is not implemented. The actual implementation supporting filename(s) after the switch is much cleaner anyway, so yes, let's go with that for project files as well. 👍

Reply:
For reference here is the issue tracking this @asbjornu  - https://github.com/GitTools/GitVersion/issues/584
It may be good to reference this in the v6 commits as I imagine you'll end up removing this parameter as a whole.

Reply:
Thanks for the reference. Yes, with the new CLI (#2272), I suspect a lot of the current arguments to become obsolete.